### PR TITLE
feat: [PIE-4419]: Support for List api in case of pipeline with errors

### PR DESCRIPTION
**Pull request by smjt-h**
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35696)

### CODE REVIEW

File: 800-pipeline-service/src/main/java/io/harness/pms/pipeline/mappers/PMSPipelineDtoMapper.java [hunk 1]
```diff
@@ -152,6 +152,9 @@ public PMSPipelineSummaryResponseDTO preparePipelineSummaryForListView(
 
   private PMSPipelineSummaryResponseDTO preparePipelineSummary(
       PipelineEntity pipelineEntity, EntityGitDetails entityGitDetails) {
+    if (pipelineEntity.getIsDraft() == null) {
```
Comment (namanvermah) [inline 800-pipeline-service/src/main/java/io/harness/pms/pipeline/mappers/PMSPipelineDtoMapper.java:155]: Please write a test for this, this class is at 100% unit coverage :P

**Review by archit-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35696)* status: 2 file(s) to review. 1 of 2 approvals obtained.

File: 800-pipeline-service/src/main/java/io/harness/pms/pipeline/mappers/PMSPipelineDtoMapper.java (hunk 1 above)
Comment (smjt-h) [reply 800-pipeline-service/src/main/java/io/harness/pms/pipeline/mappers/PMSPipelineDtoMapper.java:155]: done

**Review by namanvermah — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35696#-:-N7y1fc65jD3-lToqx10:bnfp4nl)* status: 1 file(s) to review. 1 of 2 approvals obtained.

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35696)* status: :shipit: complete! ✓

**Pull request merged**